### ISSUE TITLE
Custom Block Page

### ISSUE THREAD

**Issue by Strykerx1** [AdguardTeam/AdGuardHome#6894]
### Prerequisites

- [X] I have checked the [Wiki](https://github.com/AdguardTeam/AdGuardHome/wiki) and [Discussions](https://github.com/AdguardTeam/AdGuardHome/discussions) and found no answer

- [X] I have searched other issues and found no duplicates

- [X] I want to request a feature or enhancement and not ask a question

### The problem

Option to config a custom block page so users know that a site is blocked and that it's not technical issues. 

This feature is implemented in another DNS platform at the link below. Even with the option to add your own image and text.
https://docs.controld.com/docs/blocked-query-response

### Proposed solution

Configurable block page for blocked domains. 

### Alternatives considered and additional information

Add the ControlD block page IP address to the AGH settings. This results in AGH using ControlD's default block page. 

IPv4: 23.171.240.158
IPV6: 2606:1a40:3000:4800::158

**Comment by Strykerx1** [comment]
jslawler, thanks for the quick reply. I had already found and reviewed that as an option. However that doesn't do what I am suggesting. It still needs "an HTTP server running on some IP address and serving the page in question". 

What I am suggesting is different from the current info in the KB. I am suggesting a Adguard version of what ControlD has already done, have a default AdGuard block page preconfigured with the option to customize it. 

Default Example
![d4d3324-image](https://github.com/AdguardTeam/AdGuardHome/assets/66916285/a655ff6e-e05e-4668-a045-a5e35cb3a98e)

Config for Customized Version
![8bc488d-image](https://github.com/AdguardTeam/AdGuardHome/assets/66916285/f1b59509-a026-40ab-8565-ded61253a522)

Resulting Customized Block Page
![0cd47a0-image](https://github.com/AdguardTeam/AdGuardHome/assets/66916285/27cc5d6e-ed5f-4c2b-a5c8-a9eb995f2304)

**Comment by Strykerx1** [comment]
Thanks for linking the current thread, I hadn't found that one. 

The entire thing would be significantly easier if there was a way to redirect to a normal https url (not an ip). However I'm guessing there's technical limitations for that. Something like this https://networks.vzy.io/blocked.

**Comment by fernvenue** [comment]
Here you go: [How do I show a custom block page?](https://adguard-dns.io/kb/adguard-home/faq/#customblock) Make sure you really checked the [Wiki](https://github.com/AdguardTeam/AdGuardHome/wiki) and [Discussions](https://github.com/AdguardTeam/AdGuardHome/discussions) and found no answer next time before submit an issue.

**Comment by Strykerx1** [comment]
Fernvenue, [per my previous reply above from 2 days ago](https://github.com/AdguardTeam/AdGuardHome/issues/6894#issuecomment-2040854273) " I had already found and reviewed that as an option. However that does not do what I [was] suggesting. It still needs "an HTTP server running on some IP address and serving the page in question". What I am suggesting is different from the current info in the KB." jslawler indicated that it's currently not possible due to technical limitations. 

Furthermore, "it looks like this will all be handled by https://github.com/AdguardTeam/AdGuardHome/issues/2261 once [Structured Error Data for Filtered DNS](https://ietf-wg-dnsop.github.io/draft-ietf-dnsop-structured-dns-error/draft-ietf-dnsop-structured-dns-error.html) is approved."

**Comment by fernvenue** [comment]
> I had already found and reviewed that as an option. However that does not do what I [was] suggesting. It still needs "an HTTP server running on some IP address and serving the page in question". What I am suggesting is different from the current info in the KB.

@Strykerx1 If you really reviewed issues related to this, you will see that I have answered this question in https://github.com/AdguardTeam/AdGuardHome/issues/5669#issuecomment-1492702218:

> It's not possible, because most of the sites using HTTPS now, you can't set customized pages for them just use a DNS server, you can only set customized pages for HTTP sites, but I still don't think AdGuardHome should do that by itself. For HTTP sites, You can resolve them to your web server and host your customized pages on it, you can use [NGINX](https://docs.nginx.com/nginx/admin-guide/web-server/serving-static-content/), [Traefik](https://doc.traefik.io/traefik/routing/overview/#example-with-a-file-provider), [Caddy](https://caddyserver.com/docs/quick-starts/static-files) or any other web server applications you prefer.

And that's what exactly the documentation said. So make sure you **REALLY** checked the [Wiki](https://github.com/AdguardTeam/AdGuardHome/wiki) and [Discussions](https://github.com/AdguardTeam/AdGuardHome/discussions) and found no answer next time before submit an issue, I think that will be so helpful for the dev team.

**Comment by fernvenue** [comment]
> If you set IP 23.171.240.158 as the redirect for blocked sites in AGH DNS Settings it redirects to the page below without a HTTPS ssl error.

I don't think so, and seems you directly visted 23.171.240.158 with pure HTTP, when I visit it with HTTPS:

```
~$ curl -I https://example.com --resolve example.com:443:23.171.240.158
curl: (60) SSL certificate problem: unable to get local issuer certificate
More details here: https://curl.se/docs/sslcerts.html

curl failed to verify the legitimacy of the server and therefore could not
establish a secure connection to it. To learn more about this situation and
how to fix it, please visit the web page mentioned above.
```

By the way, a lot of applications use HTTPS by default, and the browsers like Firefox, Chrome and Edge also provide HTTPS Only mode, I don't think they will respect the invalid certificates on the address you provided.

**Comment by fernvenue** [comment]
Again, that's because you directly visted 23.171.240.158 with pure HTTP, try resolve some domains to the address and use HTTPS to vist. As I said, a lot of applications use HTTPS by default, and the browsers like Firefox, Chrome and Edge also provide HTTPS Only mode, also a lot of sites use HSTS, how can you handle them?
